test/composition.cancel.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const metasync = require('..');
+const metatests = require('metatests');
+
+const wrapAsync = (
+  // Emulate Asynchronous calls
+  callback // function
+) => {
+  setTimeout(callback, Math.floor((Math.random() * 500)));
+};
+
+let fc = null;
+metatests.test('async functions composition cancel in the middle', test => {
+
+  const fn1 = (data, cb) => {
+    wrapAsync(() => {
+      cb(null, 1);
+    });
+  };
+
+  const fn2 = (data, cb) => {
+    test.strictSame(data, [1]);
+    wrapAsync(() => {
+      if (fc) fc.cancel();
+      cb(null, 2);
+    });
+  };
+
+  const fn3 = (data, cb) => {
+    wrapAsync(() => {
+      cb(null, 3);
+    });
+  };
+
+  const fn4 = (data, cb) => {
+    wrapAsync(() => {
+      cb(null, 4);
```

Comment:
Also, add test.fail('canceled function called') maybe (and in other places where assert.fail was present)?